# Bonus: Temperature Analysis I

In [1]:
import pandas as pd
from datetime import datetime as dt

In [2]:
# "tobs" is "temperature observations"
tobs_df = pd.read_csv('../sqlalchemy_challenge/Resources/hawaii_measurements.csv')
tobs_df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [5]:
# Convert the date column format from string to datetime
tobs_df['date'] = pd.to_datetime(tobs_df['date'], format='%y%m%d')
tobs_df.head()

station       date  prcp  tobs
0  USC00519397 2010-01-01  0.08    65
1  USC00519397 2010-01-02  0.00    63
2  USC00519397 2010-01-03  0.00    74
3  USC00519397 2010-01-04  0.00    76
4  USC00519397 2010-01-06   NaN    73

In [6]:
# Set the date column as the DataFrame index
tobs_df = tobs_df.set_index(tobs_df['date'])
tobs_df.head()

station       date  prcp  tobs
date                                          
2010-01-01  USC00519397 2010-01-01  0.08    65
2010-01-02  USC00519397 2010-01-02  0.00    63
2010-01-03  USC00519397 2010-01-03  0.00    74
2010-01-04  USC00519397 2010-01-04  0.00    76
2010-01-06  USC00519397 2010-01-06   NaN    73

In [7]:
# Drop the date column
tobs_df = tobs_df.drop(columns='date')
tobs_df.head()

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

### Compare June and December data across all years 

In [8]:
# get dependency
from scipy import stats

In [9]:
# Filter data for desired months
june = tobs_df[tobs_df.index.month == 6]
june.head()
december = tobs_df[tobs_df.index.month == 12]
december.head()

station  prcp  tobs
date                               
2010-12-01  USC00519397  0.04    76
2010-12-03  USC00519397  0.00    74
2010-12-04  USC00519397  0.00    74
2010-12-06  USC00519397  0.00    64
2010-12-07  USC00519397  0.00    64

In [10]:
# Identify the average temperature for June
june_avg = june['tobs'].mean()
print(f"The average temperature in June is {june_avg}.")

The average temperature in June is 74.94411764705882.


In [11]:
# Identify the average temperature for December
december_avg = december['tobs'].mean()
print(f"The average temperature in December is {december_avg}.")

The average temperature in December is 71.04152933421226.


In [12]:
# Create collections of temperature data
june_temp = june['tobs']
dec_temp = december['tobs']

In [13]:
# Run paired t-test
stats.ttest_ind(june_temp, dec_temp)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

### The average temperature variance between June & December is 3.9 degrees. The resulting pvalue is infinitesimal. Therefore, for all practical purposes, there is no meaningful difference in temperature between June & December. 